apollo-federation/src/operation/mod.rs

```diff
@@ -2636,18 +2636,23 @@ impl SelectionSet {
                 let Some(sub_selection_type) = element.sub_selection_type_position()? else {
                     return Err(FederationError::internal("unexpected error: add_at_path encountered a field that is not of a composite type".to_string()));
                 };
-                let mut selection = Arc::make_mut(&mut self.selections)
-                    .entry(ele.key())
-                    .or_insert(|| {
+                let target = Arc::make_mut(&mut self.selections);
+                let mut selection = match target.get_mut(&ele.key()) {
+                    Some(selection) => selection,
+                    None => {
                         let unnecessary_directives = op_slice_condition_directives(path);
-                        Selection::from_element(
+                        let selection = Selection::from_element(
                             element,
                             // We immediately add a selection afterward to make this selection set
                             // valid.
                             Some(SelectionSet::empty(self.schema.clone(), sub_selection_type)),
                             Some(&unnecessary_directives),
-                        )
-                    })?;
+                        )?;
+                        let key = selection.key();
+                        let _ = target.insert(selection);
```

Comment:
do we need to check if this entry already exists?

Reply:
You're right. We do need to check for this. If one exists, do we merge them together?

Reply:
since we would attempt to insert empty selection set I think we should just get the existing entry